tests/cookbook/test_literals.py

```diff
@@ -54,3 +57,28 @@ def test_literal_column_computation_apply(daft_df, service_requests_csv_pd_df):
     daft_pd_df = daft_df.to_pandas()
     service_requests_csv_pd_df["literal_col"] = "bar"
     assert_df_equals(daft_pd_df, service_requests_csv_pd_df)
+
+
+def test_cupy_literal_tensor_cast() -> None:
+    # Try to import CuPy; skip the test if not available
+    try:
+        import cupy as cp
+    except Exception:
+        pytest.skip("cupy not installed")
+
+    # Skip when CUDA device is unavailable (or errors during availability check)
+    try:
+        if not (hasattr(cp, "cuda") and cp.cuda.is_available()):
+            pytest.skip("cupy CUDA device not available")
+    except Exception:
+        pytest.skip("cupy CUDA device not available")
+
+    arr = cp.arange(5, dtype=np.int64)
+
+    # Simple 1-row DataFrame to trigger literal conversion and casting to tensor
+    df = daft.from_pydict({"row": [0]})
+    df = df.with_column("t", lit(arr).cast(DataType.tensor(DataType.int64(), (5,))))
+    result = df.collect().to_pydict()["t"][0]
+
+    expected = cp.asnumpy(arr)
+    assert np.array_equal(result, expected)
```

Comment:
We already have a test in `tests/test_datatype_inference.py` that should cover this. Let's keep this kind of test in that file.